lib/features/email/presentation/controller/single_email_controller.dart

```diff
@@ -1703,75 +1710,94 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
         mailboxDashBoardController.accountId.value!,
-        {_displayingEventBlobId!}));
+        {_displayingEventBlobId!},
+        emailId
+      ));
     }
   }
 
-  void _rejectCalendarEventAction() {
+  void _rejectCalendarEventAction(EmailId emailId) {
     if (_rejectCalendarEventInteractor == null
       || _displayingEventBlobId == null
       || mailboxDashBoardController.accountId.value == null) {
         consumeState(Stream.value(Left(CalendarEventRejectFailure())));
     } else {
       consumeState(_rejectCalendarEventInteractor!.execute(
         mailboxDashBoardController.accountId.value!,
-        {_displayingEventBlobId!}));
+        {_displayingEventBlobId!},
+        emailId
+      ));
     }
   }
 
-  void _maybeCalendarEventAction() {
+  void _maybeCalendarEventAction(EmailId emailId) {
     if (_maybeCalendarEventInteractor == null
       || _displayingEventBlobId == null
       || mailboxDashBoardController.accountId.value == null) {
         consumeState(Stream.value(Left(CalendarEventMaybeFailure())));
     } else {
       consumeState(_maybeCalendarEventInteractor!.execute(
         mailboxDashBoardController.accountId.value!,
-        {_displayingEventBlobId!}));
+        {_displayingEventBlobId!},
+        emailId
+      ));
     }
   }
 
-  void _calendarEventSuccess(CalendarEventReplySuccess success) {
-    if (currentOverlayContext != null && currentContext != null) {
-      final appLocalization = AppLocalizations.of(currentContext!);
-      if (success is CalendarEventAccepted) {
-        appToast.showToastSuccessMessage(
-          currentOverlayContext!,
-          appLocalization.youWillAttendThisMeeting);
-      } else if (success is CalendarEventMaybeSuccess) {
-        appToast.showToastSuccessMessage(
-          currentOverlayContext!,
-          appLocalization.youMayAttendThisMeeting);
-      } else if (success is CalendarEventRejected) {
-        appToast.showToastSuccessMessage(
-          currentOverlayContext!,
-          appLocalization.youWillNotAttendThisMeeting);
-      }
+  void calendarEventSuccess(CalendarEventReplySuccess success) {
+    final session = mailboxDashBoardController.sessionCurrent;
+    final accountId = mailboxDashBoardController.accountId.value;
+
+    if (session == null || accountId == null) {
+      consumeState(Stream.value(Left(StoreEventAttendanceStatusFailure(exception: NotFoundSessionException()))));
+      return;
+    }
+
+    consumeState(_storeEventAttendanceStatusInteractor.execute(
+      session,
+      accountId,
+      success.emailId,
+      success.getEventActionType()
+    ));
+  }
+
+  void _showToastMessageEventAttendanceSuccess(StoreEventAttendanceStatusSuccess success) {
+    final selectedEmail = mailboxDashBoardController.selectedEmail.value;
+    final newEmail = selectedEmail?.updateKeywords(success.updatedEmail.keywords);
+    mailboxDashBoardController.setSelectedEmail(newEmail);
+    mailboxDashBoardController.dispatchState(Right(success));
```

Comment:
it will trigger to re-build email view?

Reply:
No, it will not be re-built because we do not listen and make any changes in EmailView when ViewState is set to StoreEventAttendanceStatusSuccess.